src/RulesEngine/RulesEngine.cs

```diff
@@ -343,7 +345,7 @@ private List<RuleResultTree> ExecuteAllRuleByWorkflow(string workflowName, RuleP
 
         private string GetCompiledRulesKey(string workflowName, RuleParameter[] ruleParams)
         {
-            var key = $"{workflowName}-" + string.Join("-", ruleParams.Select(c => c.Type.Name));
+            var key = $"{workflowName}-" + string.Join("-", ruleParams.Select(c => $"{c.Name}_{c.Type.Name}"));

```

Comment:
This should fix the mentioned issue but will make the key longer. 
Maybe we can hash the ruleParameters part of the key to ensure length of string is always constrained?

Something on the lines of
```c#
var ruleParamsHash = string.Join("-", ruleParams.Select(c => $"{c.Name}_{c.Type.Name}")).GetHashCode();
var key = $"{workflowName}-" + rulrParamsHash;
```

We need workflow name in key for clearing compiled rules when a workflow is updated/deleted, as for the rest, a hash should be sufficient